packages/ra-ui-materialui/src/input/ArrayInput/ArrayClearButton.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { ButtonProps, IconButtonWithTooltip } from '../../button';
+import ClearIcon from '@mui/icons-material/PlaylistRemove';
+import { useSimpleFormIterator } from './useSimpleFormIterator';
+
+export const ArrayClearButton = (props: ButtonProps) => {
+    const { clear } = useSimpleFormIterator();
+
+    return (
+        <IconButtonWithTooltip
+            label="ra.action.clear_array_input"
+            size="small"
+            onClick={() => clear()}
```

Comment:
```suggestion
            onClick={clear}
```

To be double check, but I think passing the function variable itself is enough.